File: solutions/_881_boats_to_save_people/solution.java
package _881_boats_to_save_people;

import java.util.Arrays;

class Solution {

  public int numRescueBoats(int[] people, int limit) {
    Arrays.sort(people);
    int boats = 0;
    int i = 0;
    int j = people.length - 1;

    while (i <= j) {
      if (people[i] + people[j] <= limit) {
        i++;
      }
      j--;
      boats++;
    }

    return boats;
  }
}
/*
https://leetcode.com/problems/boats-to-save-people/
*/

File: solutions/_881_boats_to_save_people/solution.js
/**
 * @param {number[]} people
 * @param {number} limit
 * @return {number}
 */
var numRescueBoats = function (people, limit) {
  people.sort((a, b) => a - b);
  let boats = 0;
  let i = 0;
  let j = people.length - 1;

  while (i <= j) {
    if (people[i] + people[j] <= limit) {
      i++;
    }
    j--;
    boats++;
  }

  return boats;
};

module.exports = { numRescueBoats };

/*
https://leetcode.com/problems/boats-to-save-people/
*/
